### ISSUE TITLE
QColor is rising an error when it's inside a QMenu and you double click on it

### ISSUE THREAD

**Issue by kevingermain** [quasarframework/quasar#10028]
If QColor is inside a QMenu and the user double click (by mistake for example) QColor rise an error `Uncaught TypeError: Cannot read property 'clientWidth' of null` and it's closing the menu even it's not the desired behavior.

Can be simply tested here: https://quasar.dev/vue-components/color-picker#with-qinput
Open the console first, try to double click, and see the error and the QMenu closing unexpectedly.

(it works if the screen is small enough (mobile mode for example))

![Screenshot at 18-11-17](https://user-images.githubusercontent.com/1334099/125156738-67608200-e1aa-11eb-801e-afdc378becb9.png)

**Comment by hawkeye64** [comment]
@kevingermain I am not seeing any issues here.
![image](https://user-images.githubusercontent.com/10262924/125331017-b02c4e00-e304-11eb-84d7-bfd8dc540235.png)

Can you try in incognito mode?

**Comment by kevingermain** [comment]
@hawkeye64 I tried with all disabled extensions, with brave (without the shield) and another chrome from another computer, still the issue (the issue is not showing with firefox).

Try to double click a lot on the same spot.

**Comment by MilosPaunovic** [comment]
After 10 minutes od relentles clicking, I've managed to reproduce the issue:

- Open color picker
- Click on any color and hold down the mouse button
- Swipe around selecting colors (don't let mouseup event occur)
- Quickly double click outside of picker

Repro video [link](https://www.dropbox.com/s/7l2h4k2kxr4jgec/Screen%20Recording%202021-07-15%20at%2014.12.37.mov?dl=0).

It raise error in console:
![Screenshot 2021-07-15 at 14 14 09](https://user-images.githubusercontent.com/62475782/125786300-c339c42c-6e24-4cdf-9b41-bc458629ad25.png)

**Comment by hawkeye64** [comment]
I am still unable to reproduce it. But, I have an idea what is happening. The mouse event queue is getting backed up and by the time you click outside, the panel has been destroyed, but mouse events are still coming in and being processed.

```js
    function changeSpectrum (left, top, change) {
      const panel = spectrumRef.value
      if (panel === void 0) { return }

      const
        width = panel.clientWidth, // < -- here
        height = panel.clientHeight,
        rect = panel.getBoundingClientRect()
```

**Comment by rstoenescu** [comment]
Fix will be available in Quasar v2.0.3

**Comment by kevingermain** [comment]
The error disappear in the console but the problem still here. Can someone confirm please.
